cli/send_test.go

```diff
@@ -65,7 +67,51 @@ func TestSendCLI(t *testing.T) {
 			mockSrvcs.EXPECT().Close(),
 		)
 		err := app.Run([]string{"lotus", "send", "t01", "1"})
-		assert.NoError(t, err)
-		assert.EqualValues(t, sigMsg.Cid().String()+"\n", buf.String())
+		require.NoError(t, err)
+		require.EqualValues(t, sigMsg.Cid().String()+"\n", buf.String())
+	})
+}
+
+func TestSendEthereum(t *testing.T) {
+	oneFil := abi.TokenAmount(types.MustParseFIL("1"))
+
+	t.Run("simple", func(t *testing.T) {
+		app, mockSrvcs, buf, done := newMockApp(t, sendCmd)
+		defer done()
+
+		testEthAddr, err := ethtypes.CastEthAddress(make([]byte, 20))
+		require.NoError(t, err)
+		testAddr := mustAddr(testEthAddr.ToFilecoinAddress())
+
+		params := abi.CborBytes([]byte{1, 2, 3, 4})
+		var paramsBuf bytes.Buffer
+		require.NoError(t, params.MarshalCBOR(&paramsBuf))
+
+		arbtProto := &api.MessagePrototype{
+			Message: types.Message{
+				From:   testAddr,
+				To:     mustAddr(address.NewIDAddress(1)),
+				Value:  oneFil,
+				Method: builtin.MethodsEVM.InvokeContract,
+				Params: paramsBuf.Bytes(),
+			},
+		}
+		sigMsg := fakeSign(&arbtProto.Message)
+
+		gomock.InOrder(
+			mockSrvcs.EXPECT().MessageForSend(gomock.Any(), SendParams{
+				From:   testAddr,
+				To:     mustAddr(address.NewIDAddress(1)),
+				Val:    oneFil,
+				Method: builtin.MethodsEVM.InvokeContract,
+				Params: paramsBuf.Bytes(),
+			}).Return(arbtProto, nil),
+			mockSrvcs.EXPECT().PublishMessage(gomock.Any(), arbtProto, false).
+				Return(sigMsg, nil, nil),
+			mockSrvcs.EXPECT().Close(),
+		)
+		err = app.Run([]string{"lotus", "send", "--from-eth-addr", testEthAddr.String(), "--params-hex", "01020304", "f01", "1"})
```

Comment:
Note: this _won't_ work if the the default address is an eth address, because the command won't see that. Is that something we want to support?

Reply:
(nope? lotus doesnt support import eth address to wallet yet, would be nice to have later for us to test but out of scope of this pr imho